src/runtime/link-accounts-flow.js

```diff
@@ -271,24 +271,38 @@ export class LinkSaveFlow {
         });
       }
     });
+    let linkConfirm = null;
     /** {!Promise<boolean>} */
     return this.dialogManager_.openView(this.activityIframeView_,
         /* hidden */ true).then(() => {
-          return this.activityIframeView_.port().then(port => {
-            return acceptPortResultData(
-                port,
-                feOrigin(),
-                /* requireOriginVerified */ true,
-                /* requireSecureChannel */ true);
-          }).then(result => {
-            return result['linked'];
-          }).catch(() => {
-            return false;
-          }).then(result => {
-            // The flow is complete.
-            this.dialogManager_.completeView(this.activityIframeView_);
-            return result;
+        return this.activityIframeView_.port().then(port => {
+          return acceptPortResultData(
+              port,
+              feOrigin(),
+              /* requireOriginVerified */ true,
+              /* requireSecureChannel */ true);
+        }).then(result => {
+          // The flow is complete.
+          if (!result['linked']) {
+            throw new Error('save subscription failed');
```

Comment:
This shouldn't be an error. User refusing to link should be transparent to the publisher.

Reply:
You are right, we should just return here.